models/demos/t3000/mixtral8x7b/tests/test_mixtral_model_prefill.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+import os
+import torch
+import pytest
+import numpy as np
+from loguru import logger
+from sklearn.metrics import top_k_accuracy_score
+import time
+
+# Set Mixtral flags for CI, if CI environment is setup
+if os.getenv("CI") == "true":
+    os.environ["MIXTRAL_CKPT_DIR"] = "/mnt/MLPerf/tt_dnn-models/Mistral/Mixtral-8x7B-v0.1/"
+    os.environ["MIXTRAL_TOKENIZER_PATH"] = "/mnt/MLPerf/tt_dnn-models/Mistral/Mixtral-8x7B-v0.1/"
+    os.environ["MIXTRAL_CACHE_PATH"] = "/mnt/MLPerf/tt_dnn-models/Mistral/Mixtral-8x7B-v0.1/"
+    os.environ["TT_METAL_ASYNC_DEVICE_QUEUE"] = "1"
+    os.environ["WH_ARCH_YAML"] = "wormhole_b0_80_arch_eth_dispatch.yaml"
+
+import ttnn
+from ttnn import ReplicateTensorToMesh, ConcatMeshToTensor
+
+from models.demos.t3000.mixtral8x7b.tt.mixtral_common import (
+    prepare_inputs_ttnn_prefill,
+    prepare_rotation_mat_ttnn,
+    get_rot_transformation_mat,
+)
+from models.demos.t3000.mixtral8x7b.tt.mixtral_model import TtTransformer
+from models.demos.t3000.mixtral8x7b.reference.model import Transformer
+from models.demos.t3000.mixtral8x7b.reference.tokenizer import Tokenizer
+from models.demos.t3000.mixtral8x7b.tt.model_config import TtModelArgs
+from models.utility_functions import comp_pcc, comp_allclose
+
+
+class Emb(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.emb = torch.nn.Embedding(32000, 4096)
+
+    def forward(self, x):
+        return self.emb(x)
+
+
+@pytest.mark.parametrize(
+    "seq_len",
+    (128, 1024, 2048),
+)
+@pytest.mark.parametrize(
+    "n_layers",
+    (32,),
```

Comment:
I've added a test for 1L as well. We can then choose which pipelines we want to run this test.